app/assets/javascripts/discourse/app/static/prosemirror/core/plugin.js

```diff
@@ -0,0 +1,56 @@
+import { Plugin } from "prosemirror-state";
+
+export function extractNodeViews(extensions) {
+  return extensions.reduce((acc, { nodeViews }) => {
+    if (nodeViews) {
+      Object.entries(nodeViews).forEach(([name, NodeViewClass]) => {
+        acc[name] = (node, view, getPos) =>
+          new NodeViewClass(node, view, getPos);
+      });
+    }
+
+    return acc;
+  }, {});
+}
+
+export function extractPlugins(extensions, params, view) {
+  return (
+    extensions
+      .reduce((acc, extension) => {
+        if (extension.plugins instanceof Array) {
+          acc.push(...extension.plugins);
+        } else if (extension.plugins) {
+          acc.push(extension.plugins);
+        }
+        return acc;
+      }, [])
```

Comment:
I think this is the same?
```suggestion
      .flatMap((acc, extension) => extension.plugins)
```